Further iteration on the sidebar design

j-f1 opened the pull request:
<details><summary>Outdated screenshots</summary>

<img width="249" align=left alt="Screenshot_2023-01-10 16 31 58" src="https://user-images.githubusercontent.com/25517624/211666802-90693583-b8e4-4002-b9de-1ae200297a90.png">
<img width="343" align=right alt="Screenshot_2023-01-10 16 02 51" src="https://user-images.githubusercontent.com/25517624/211662042-81fa54ec-90ce-4628-91fc-cbac537caf8f.png">

&larr; The tools are back inline. The section defaults to closed unless you load the page into one of the tools, then it will open itself.

…and the parent item highlights itself when collapsed if any of its children are selected &rarr;

.

.


<img width="256" align=right alt="Screenshot_2023-01-10 16 04 40" src="https://user-images.githubusercontent.com/25517624/211662290-244195fd-c3b1-47ab-b469-dec0237f1722.png">

<br>New look for the Accounts section: &rarr;

- The “Accounts” header now toggles the visibility of the entire section, just like the Tools section (although the Accounts section is always open by default)
- There’s now a new header called “All accounts” (or “All open accounts” if you have closed accounts, would appreciate feedback on this distinction) that shows your net worth, and lets you see transaction history for on and off budget accounts combined (i.e. what “Accounts” used to do)
- The button to add an account is now at the bottom of the accounts list, below “Closed accounts.” While it could theoretically scroll out of view, by the time you make that many accounts you’ll hopefully know where to look, right? (or you’ll end up scrolling to the bottom of the list anyway)

Here’s how it looks with both open: &darr;
<img width="241" alt="Screenshot_2023-01-10 16 33 10" src="https://user-images.githubusercontent.com/25517624/211667033-e66c6bac-8326-496f-9e3e-ca3d8cba863a.png">

MatissJanis commented:
👋 I really dislike the big gap between Schedules -> Tools -> Accounts. How would it look like if we kept the spacing between all these items aligned?

For the sub-menu: could we increase padding from the left side to better signify the sub-items? Here's some inspiration: https://dribbble.com/shots/18288223-Sidebar-navigation-Untitled-UI

j-f1 commented:
I assume that James didn’t have indentation for the accounts section so that long account names could be more legible, which is why I’m hesitant to do it now. However, maybe that isn’t a huge problem (since you can always see the full name by clicking on the account, and you can also rename the account to be more distinctive)

MatissJanis commented:
> I assume that James didn’t have indentation for the accounts section so that long account names could be more legible, which is why I’m hesitant to do it now. However, maybe that isn’t a huge problem (since you can always see the full name by clicking on the account, and you can also rename the account to be more distinctive)

Maybe we can indent only the new sub-menu items and keep accounts as-is? It will create a bit of inconsistency, but perhaps that's find in this case.. what do you think?

MatissJanis commented:
One more piece of feedback I received in private from another member of the community:

>  I am not too sure about the size of the gap between All Accounts and Budget accounts either.

This one makes sense too IMO. If we cold remove the gap here that would be nicer.

<img width="248" alt="Screenshot 2023-01-10 at 22 53 42" src="https://user-images.githubusercontent.com/886567/211679425-b5519ccf-61b1-4a35-a220-20f521ea96ce.png">

Kidglove57 commented:
Thank you @j-f1 much appreciated. All functions well and glad to see the dropdown under Tools!
I am with you @MatissJanis re the big gap between Schedules -> Tools -> Accounts. I am one of the few who have loads of accounts - maybe 20 (always chasing the best deal!) thus  needing as much space in the sidebar as possible. So from my point of view the extra “all accounts” line is a mixed blessing.

j-f1 commented:
Removing the spacing between the accounts feels pretty noisy to me:
<img width="243" alt="Screenshot_2023-01-10 19 03 43" src="https://user-images.githubusercontent.com/25517624/211687604-3d665362-cb0e-4ab5-a49b-d2c0bae23501.png">

j-f1 commented:
Here it is with the indent:
<img width="240" alt="Screenshot_2023-01-10 19 07 13" src="https://user-images.githubusercontent.com/25517624/211687923-71ba3f54-9ae3-48a5-9118-4e16104c7c6b.png">

trevdor commented:
🤣 
> by the time you make that many accounts you’ll hopefully know where to look, right? 

First I'll add my thanks overall, @j-f1. Big fan of your idea to move `+ New Account` to bottom. 

I'll be ok with however we land this, but a couple more personal opinions for you:

1. I wouldn't get a lot of value out of quick-glance net worth at `All open accounts`. I wish my net worth grew faster (or grew at all?), but I don't mind going here for the same value:
<img width="447" alt="all open accounts value" src="https://user-images.githubusercontent.com/5862724/211708643-980990a9-53e6-4bb3-92a9-51b590d958e0.png">

2. I don't ever imagine wanting to collapse my accounts, though I've floated the sidebar away completely from time to time. Following from my other suggestion: if we drop `All open accounts`, the gap goes away and clicking the Accounts heading can keep doing what it does in production today. It ends up:

![sidebar_noAllAccounts](https://user-images.githubusercontent.com/5862724/211709437-9cdcf052-458e-4cea-9731-b682556a92d8.png)

Teprifer commented:
I'll add a +1 with comment to Trev's note above about doing away with an all open accounts total, for those of us with no off-budget accounts, it would just be needless duplication of the same value.

j-f1 commented:
> 2\. I don't ever imagine wanting to collapse my accounts

One reason I’ve thought that this could be valuable would be if I don’t want someone shoulder-surfing to be able to see all of my account balances (although that could also be remedied by simply hiding the sidebar). It also felt a bit weird to me to have two very similar items in the sidebar (namely the Tools and Accounts headings) have completely different behavior. Maybe the disclosure arrow will help alleviate that, but I don’t want the “All accounts” listing to be missed by people who could gain value from it. Today, it’s not entirely clear that the option exists at all, and moving the “Add account button” away is part of making that more obvious.

trevdor commented:
> It also felt a bit weird to me to have two very similar items in the sidebar (namely the Tools and Accounts headings) have completely different behavior. 

imo, it's ok if Tools is the only item that expands. But I see your point. 
Although, maybe having Tools in the menu just isn't a perfect fit for the nav paradigm.

> I don’t want the “All accounts” listing to be missed by people who could gain value from it. Today, it’s not entirely clear that the option exists at all

Agree--it wasn't obvious to me. Although, tbh, it took me until your PR preview build to learn I could see all `For budget` accounts by clicking that label...at which point "All accounts" made more sense. So...🤷🏻‍♂️🤷🏻‍♂️🤷🏻‍♂️ 

I'm still a vote for not collapsing Accounts, but more ambivalent about "All accounts". And ultimately think we can't go all that wrong whatever we choose.

Y'all can duke it out in the parking lot. 😄 👊🏻

Kidglove57 commented:
@trevdor has nicely summed up my personal preferences too. 
1) keep ”all accounts” where it currently is I.e. see the total and transactions by clicking on the “Accounts” line as now.
2) could then retain the current spacing between Accounts, Budgeted and Off Budget
3) I am ambivalent about moving the Tools to the sidebar but if it is to go there then overall I prefer the latest iteration.
4) I am happy for the collapse accounts functionality to be there for those who need it. 
5) Very happy to have Add Account at the bottom

I click Accounts (All Accounts!) often to search for a transaction. Also to bulk edit (add a note to a number of transactions at once for example). I also use this All Accounts view to filter for end of year tax data. So I agree that it is nice for it to be clear that it is there.

Having said that so much in Actual is neatly hidden away - the expansion of an Account heading to show uncleared, the facility to edit an account name or add an account note as examples. You have to know they are there or you might never spot them. I have no issue with this but it makes me wonder how far we want to go in the opposite direction ….

j-f1 commented:
@Kidglove57 I’m not sure how to square these 2 things:

- keep ”all accounts” where it currently is I.e. see the total and transactions by clicking on the “Accounts” line as now.
- I am happy for the collapse accounts functionality to be there for those who need it.

…how could the collapse functionality be accessed if clicking on “Accounts” opens the all accounts screen?

Kidglove57 commented:
Very good point!! I admit I was writing somewhat carelessly as someone who will not use the collapse accounts option.

j-f1 commented:
What about getting rid of the header entirely? (Also trying replacing the gray headers with bold so they still feel clickable)

<img width="254" alt="Screenshot_2023-01-11 12 40 10" src="https://user-images.githubusercontent.com/25517624/211878684-785b9926-5204-48af-82ea-987a87a32f02.png">

MatissJanis commented:
Just my 2 cents:

Of all the versions we had, I liked this one the best:
- account list collapsable
- account title with the same font/size as other primary nav items
- gray sub-nav items
- "add accounts" and "closed accounts" links in gray
- consistent spacing between nav items

![img](https://user-images.githubusercontent.com/5862724/211709437-9cdcf052-458e-4cea-9731-b682556a92d8.png)

Kidglove57 commented:
I would be extremely happy with either of the two immediately previous options! My preference (solely from a looks rather than functionality perspective) being for what @MatissJanis has illustrated

j-f1 commented:
@MatissJanis Where would you click to show all accounts at once in that UI?

MatissJanis commented:
> @MatissJanis Where would you click to show all accounts at once in that UI?

I liked the extra row you added for "all accounts" (which could be used to access all account data). But I disliked the big gap between "all accounts" and "for budget".

Kidglove57 commented:
Agree about the gap. 

One question (excuse ignorance)- why would the line need to be called "allOpen accounts"?
(1) Closed accounts would surely all have a nil balance and therefore not effect net worth and 
(2)I would still want to be able to include historic transactions in closed accounts for filter/searching

rich-howell commented:
The only thing I would say is **Payees** and **Rules** are not really tools, the heading itself seems oddly named. Other than that the design is nice.

j-f1 commented:
> One question (excuse ignorance) - why would the line need to be called "All Open Accounts"?
> (1) Closed accounts if included would surely all have a nil balance and therefore not effect net worth and
> (2) I would still want to be able to include the historic transactions in closed accounts for filtering/searching

Ah yep, it is actually “All accounts.” Fixing!

j-f1 commented:
Latest batch of changes: Renamed “Tools” to “More” and changed the icon to a disclosure chevron, added a divider to fill up the space between the main nav items and the accounts listing.

<img width="238" alt="Screenshot_2023-01-11 15 23 35" src="https://user-images.githubusercontent.com/25517624/211910127-2def102e-bbe1-4ec9-b2ac-fb049b837b86.png"><details><summary>Expand for tall screenshot</summary>
<img width="242" alt="Screenshot_2023-01-11 15 24 21" src="https://user-images.githubusercontent.com/25517624/211910174-c5102a7f-c469-4c9c-80a3-a4536cdee43e.png">

MatissJanis reviewed (approved):
Amazing! Love the version we have now.

I'll leave the PR open for a bit longer for the other folks to check out the changes and give their feedback. But IMO we're good to go with this.

Kidglove57 commented:
I love what this has become - thank you!

j-f1 commented:
Just pushed a couple of changes mostly in the code organization but also to match up the spacing between all the different sections in the sidebar (they’re all 15px now)

trevdor reviewed (approved):
I think you win a medal for most patience with PR feedback. 🥇 
But it's a high-profile bit of the app that affects everyone, so thanks for being so open to adapting and trying out all sorts of ideas.

j-f1 commented:
Thank you for all the feedback! The result is definitely much better after the changes :)

MatissJanis commented:
Thanks for the amazing work Jed! Keep the PRs coming :)
